api/src/main/java/io/druid/timeline/partition/NoneShardSpec.java

```diff
@@ -32,6 +33,11 @@
  */
 public class NoneShardSpec implements ShardSpec
 {
+  private final static NoneShardSpec INSTANCE = new NoneShardSpec();
```

Comment:
we could even make the constructor private to make sure instance is always used.